diem-node/src/lib.rs

```diff
@@ -415,6 +435,32 @@ pub fn setup_environment(node_config: &NodeConfig, logger: Option<Arc<Logger>>)
     // TODO set up on-chain discovery network based on UpstreamConfig.fallback_network
     // and pass network handles to mempool/state sync
 
+    // For now, spawn all of the storage-service servers on the same runtime.
+    let storage_service_rt = Builder::new_multi_thread()
+        .thread_name("storage-service-servers")
+        .enable_all()
+        .build()
+        .expect("Failed to start the DiemNet storage-service runtime.");
+    let storage_reader = StorageReader::new(Arc::clone(&db_rw.reader));
+    for storage_service_events in storage_service_server_network_handles {
+        let storage_service_server = StorageServiceServer::new(
```

Comment:
Yeah, it's kinda annoying that we need a new server for each network, but there's probably not much we can do for now.